app/mci/cmd/delete.go

```diff
@@ -0,0 +1,152 @@
+			}
+		},
+	}
+	AddDeleteFlags(cmd, &options)
+	return cmd
+}
+
+func AddDeleteFlags(cmd *cobra.Command, options *DeleteOptions) error {
+	cmd.Flags().StringVarP(&options.IngressFilename, "ingress", "i", options.IngressFilename, "filename containing ingress spec")
+	cmd.Flags().StringVarP(&options.Kubeconfig, "kubeconfig", "k", options.Kubeconfig, "path to kubeconfig")
+	cmd.Flags().StringVarP(&options.GCPProject, "gcp-project", "", options.GCPProject, "name of the gcp project")
+	// TODO Add a verbose flag that turns on glog logging.
+	return nil
+}
+
+func ValidateDeleteArgs(options *DeleteOptions, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("unexpected args: %v. Expected one arg as name of load balancer.", args)
+	}
+	// Verify that the required params are not missing.
+	if options.IngressFilename == "" {
+		return fmt.Errorf("unexpected missing argument ingress.")
+	}
+	if options.GCPProject == "" {
+		return fmt.Errorf("unexpected missing argument gcp-project.")
+	}
+	return nil
+}
+
+func RunDelete(options *DeleteOptions, args []string) error {
+	options.LBName = args[0]
+
+	// Unmarshal the YAML into ingress struct.
+	var ing v1beta1.Ingress
+	if err := unmarshall(options.IngressFilename, &ing); err != nil {
+		return fmt.Errorf("error in unmarshalling the yaml file %s, err: %s", options.IngressFilename, err)
+	}
+	clientset, err := getClientset(options.Kubeconfig)
+	if err != nil {
+		return fmt.Errorf("unexpected error in instantiating clientset: %v", err)
+	}
+	cloudInterface, err := cloudinterface.NewGCECloudInterface(options.GCPProject)
+	if err != nil {
+		return fmt.Errorf("error in deleting cloud interface: %s", err)
+	}
+
+	// Delete ingress in all clusters.
+	if err := deleteIngress(options.Kubeconfig, options.IngressFilename); err != nil {
+		return err
+	}
+
+	lbs := gcplb.NewLoadBalancerSyncer(options.LBName, clientset, cloudInterface)
+	return lbs.DeleteLoadBalancer(&ing)
+}
+
+// Extracts the contexts from the given kubeconfig and deletes ingress in those context clusters.
+func deleteIngress(kubeconfig, ingressFilename string) error {
+	// TODO(nikhiljindal): Allow users to specify the list of clusters to delete the ingress in
+	// rather than assuming all contexts in kubeconfig.
+	clusters, err := getClusters(kubeconfig)
+	if err != nil {
+		return err
+	}
+	return deleteIngressInClusters(kubeconfig, ingressFilename, clusters)
+}
+
+// Deletes the given ingress in the given list of clusters.
+func deleteIngressInClusters(kubeconfig, ingressFilename string, clusters []string) error {
+	kubectlArgs := []string{"kubectl"}
+	if kubeconfig != "" {
+		kubectlArgs = append(kubectlArgs, fmt.Sprintf("--kubeconfig=%s", kubeconfig))
+	}
+	// TODO(nikhiljindal): Validate and optionally add the gce-multi-cluster class annotation to the ingress YAML spec.
+	deleteArgs := append(kubectlArgs, []string{"delete", fmt.Sprintf("--filename=%s", ingressFilename)}...)
```

Comment:
Will update this to use the go client instead of kubectl when https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/12 merges.